application/backend/app/services/configuration_service.py

```diff
@@ -45,69 +47,60 @@ def _notify_source_changed(self) -> None:
         with self._config_changed_condition:
             self._config_changed_condition.notify_all()
 
-    @staticmethod
-    def _on_config_changed(config_id: UUID, field: PipelineField, db: Session, notify_fn: Callable[[], None]) -> None:
+    def _on_config_changed(self, config_id: UUID, field: PipelineField, notify_fn: Callable[[], None]) -> None:
         """Notify threads or child processes that the configuration has changed.
         Notification triggered only when the configuration is used by the active pipeline."""
-        pipeline_repo = PipelineRepository(db)
+        pipeline_repo = PipelineRepository(self._db_session)
         active_pipeline = pipeline_repo.get_active_pipeline()
         if active_pipeline and getattr(active_pipeline, field) == str(config_id):
             notify_fn()
 
     def list_sources(self) -> list[Source]:
-        return self._source_service.list_all()
+        return self._source_service.list_all(self._db_session)
```

Comment:
The source and sink services receive the session through the methods (`list_all`, `get_by_id`, ...), while other services like `ConfigurationService` and `DatasetService` get the session through the constructor (`__init__`). Is it by design? Can't we pass the session to the `GenericPersistenceService` when instantiating it?